### PR TITLE
FE-7857: Updated Kraken with latest Rotation Order changes

### PR DESCRIPTION
**Pull request by EricTRocks**
From Fabric Rotation Order changes.

The changes to the rotation orders in Fabric mean that the use of the Rotation orders in Kraken need to be updated in both the Euler and RotationOrder Math objects but also the use of them in the Plugin builders.

### CODE REVIEW

File: Python/kraken/core/maths/rotation_order.py [hunk 1]
```diff
@@ -100,47 +100,47 @@ def set(self, order):
...
-        newOrder = 0
+        newOrder = 4
 
         if type(order) == str:
             lowerOrder = order.lower()
             if lowerOrder == 'xyz':
-                newOrder = 0
+                newOrder = 4
             elif lowerOrder == 'yzx':
                 newOrder = 1
```
Comment (jcgagnon) [inline Python/kraken/core/maths/rotation_order.py]: Doesn't these have to be changed too? See rotationOrderStrToIntMapping above

File: Python/kraken/plugins/max_plugin/builder.py [hunk 2]
```diff
@@ -2222,14 +2222,29 @@ def setTransform(self, kSceneItem):
...
+
+        # Max's rotation order enums:
+        # 1 XYZ
+        # 2 XZY
+        # 3 YZX
+        # 4 YXZ
+        # 5 ZXY
+        # 6 ZYX
         rotOrderRemap = {
-                0: 1,
```
Comment (jcgagnon) [inline Python/kraken/plugins/max_plugin/builder.py]: Although what used to be "XYZ" is now named "ZYX", the behavior of each order "integer" has not changed. From that, I'd expect that mapping from "integer" order from Max to Fabric to not change, unless there was a bug before.

File: Python/kraken/plugins/maya_plugin/builder.py [hunk 3]
```diff
@@ -579,18 +579,30 @@ def buildOrientationConstraint(self, kConstraint, buildName):
...
                 # 2 ZXY
                 # 3 XZY
-                # 4 YXZ <-- 5 in Fabric
-                # 5 ZYX <-- 4 in Fabric
+                # 4 YXZ
+                # 5 ZYX
+
                 order = kConstraint.getConstrainee().ro.order
                 if order == 4:
-                    order = 5
```
Comment (jcgagnon) [inline Python/kraken/plugins/maya_plugin/builder.py]: Same comment as above for Max; I wouldn't expect any change here, unless you're fixing a bug.

File: Python/kraken/plugins/maya_plugin/builder.py [hunk 4]
```diff
@@ -678,27 +690,35 @@ def buildPoseConstraint(self, kConstraint, buildName):
...
                 # 3 XZY
-                # 4 ZYX
-                # 5 YXZ
+                # 4 YXZ
+                # 5 ZYX
 
                 rotOrderRemap = {
-                    0: 4,
+                    0: 5,
                     1: 3,
```
Comment (jcgagnon) [inline Python/kraken/plugins/maya_plugin/builder.py]: Same here; on the "integer side", the behavior didn't change

File: Python/kraken/plugins/si_plugin/builder.py [hunk 5]
```diff
@@ -592,12 +592,12 @@ def buildOrientationConstraint(self, kConstraint, buildName):
             # It appears Softimage uses the reversed orders
             # Not the same orders.
             rotOrderRemap = {
-                0: 0,
-                1: 3,
-                2: 4,
-                3: 1,
-                4: 5,
```
Comment (jcgagnon) [inline Python/kraken/plugins/si_plugin/builder.py]: Same here, I wouldn't expect any integer remap change, unless the previous behavior was wrong.

File: Python/kraken/plugins/si_plugin/builder.py [hunk 6]
```diff
@@ -662,12 +662,12 @@ def buildPoseConstraint(self, kConstraint, buildName):
             # 5 ZYX
 
             rotOrderRemap = {
-                0: 4,
+                0: 5,
                 1: 1,
                 2: 2,
```
Comment (jcgagnon) [inline Python/kraken/plugins/si_plugin/builder.py]: idem...

File: Python/kraken/plugins/si_plugin/builder.py [hunk 7]
```diff
@@ -1432,12 +1432,12 @@ def setTransform(self, kSceneItem):
 
         # Softimage's rotation orders remapped:
         rotOrderRemap = {
-            0: 0,
-            1: 3,
-            2: 4,
-            3: 1,
-            4: 5,
-            5: 2
```
Comment (jcgagnon) [inline Python/kraken/plugins/si_plugin/builder.py]: idem...

**Comment by EricTRocks** [comment]
Things got a bit wonky with additional changes I made to this branch since the review. Closing this and opening a new one.

**Pull request closed without merging**